### PR TITLE
chore(nx): update nx dependencies to latest version v15.3.0

### PR DESCRIPTION
**Pull request by overbit**
Close: #4521 

## PR Details

Upgraded monorepo framework to latest version

## PR Checklist

- [ ] Tests for the changes have been added
- [ ] `npm test` doesn't throw any error

IMPORTANT: Please review the [CONTRIBUTING.md](https://github.com/amplication/amplication/blob/master/CODE_OF_CONDUCT.md) file for detailed contributing guidelines.

### CODE REVIEW

**Comment by abrl91** [comment]
Does it something that can work for us? (for the client)
https://youtu.be/KBFQZw5ynFs?t=314

**Comment by overbit** [comment]
> Does it something that can work for us? (for the client) https://youtu.be/KBFQZw5ynFs?t=314

We already manually fully migrated the client to nx. In future we could try that approach for the `cli` and  `dsg` @abrl91

File: libs/util/nestjs/kafka/project.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
-  "$schema": "../../../../node_modules/nx/schemas/project-schema.json",
   "name": "util-nestjs-kafka",
+  "$schema": "../../../../node_modules/nx/schemas/project-schema.json",
```
Comment (barshimi) [inline libs/util/nestjs/kafka/project.json:3]: is there a way to create alias to node module, instead of using that root ?

Comment (overbit) [reply libs/util/nestjs/kafka/project.json:3]: Potentially, but this transformation has been performed by nx migration script so I would avoid any change for now.

**Review by yuval-hazaz — APPROVED** [review]

**Pull request merged**